example-extension/Classes/Math.php

```diff
@@ -0,0 +1,13 @@
+<?php
+
+declare(strict_types=1);
+
+namespace ExampleVendor\ExampleExtension;
+
+class Math
+{
+    public function add(int|float $x, int|float $y): float|int
+    {
+        return $x + $y;
+    }
```

Comment:
@mkarulin This is by no means important, but I think this would be a good opportunity to learn more about templates. I haven't looked it up properly and thusly this is probably not 100% correct, but you should be able to add suchlike annotation for PHPStan:

```
/**
 * @template T of int|float
 * @param T $x
 * @param T $y
 * @return T
 */
```

With that you could say that you expect both input parameters to have the same type, and because of that you can better predict the return type, since int+int=int and float+float=float. PHPStan will consequently "ensure" that the method is not called with two different input types, e.g. int+float. By "ensure" I mean that it will fail its checks on code that is calling the method incorrectly.